File: 351004/Levental/task320-rest/src/main/java/by/bsuir/task320/mapper/CreatorMapper.java
package by.bsuir.task320.mapper;

import by.bsuir.task310.domain.Creator;
import by.bsuir.task310.dto.request.CreatorRequestTo;
import by.bsuir.task310.dto.response.CreatorResponseTo;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface CreatorMapper {
    Creator toEntity(CreatorRequestTo dto);
    CreatorResponseTo toResponse(Creator entity);
}

File: 351002/Abrashin/publisher/src/main/kotlin/com/example/distcomp/mapper/CreatorMapper.kt
package com.example.distcomp.mapper

import com.example.distcomp.dto.request.CreatorRequestTo
import com.example.distcomp.dto.response.CreatorResponseTo
import com.example.distcomp.model.Creator
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface CreatorMapper {
    fun toEntity(request: CreatorRequestTo): Creator
    fun toResponse(entity: Creator): CreatorResponseTo
}
